card/src/test/java/com/adyen/checkout/card/CardComponentTest.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by josephj on 7/12/2022.
+ */
+
+package com.adyen.checkout.card
+
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.viewModelScope
+import app.cash.turbine.test
+import com.adyen.checkout.action.DefaultActionHandlingComponent
+import com.adyen.checkout.action.GenericActionDelegate
+import com.adyen.checkout.components.PaymentComponentEvent
+import com.adyen.checkout.components.ui.ViewProvider
+import com.adyen.checkout.components.ui.view.ComponentViewType
+import com.adyen.checkout.test.TestDispatcherExtension
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@ExtendWith(MockitoExtension::class, TestDispatcherExtension::class)
+class CardComponentTest(
+    @Mock private val cardDelegate: CardDelegate,
+    @Mock private val genericActionDelegate: GenericActionDelegate,
+    @Mock private val actionHandlingComponent: DefaultActionHandlingComponent,
+) {
+
+    private lateinit var component: CardComponent
+
+    @BeforeEach
+    fun before() {
+        whenever(cardDelegate.viewFlow) doReturn MutableStateFlow(CardComponentViewType)
+        whenever(genericActionDelegate.viewFlow) doReturn MutableStateFlow(null)
+
+        component = CardComponent(
+            cardDelegate,
+            genericActionDelegate,
+            actionHandlingComponent,
+        )
+    }
+
+    @Test
+    fun `when component is created then delegates are initialized`() {
+        verify(cardDelegate).initialize(component.viewModelScope)
+        verify(genericActionDelegate).initialize(component.viewModelScope)
+    }
+
+    @Test
+    fun `when component is cleared then delegates are cleared`() {
+        component.onCleared()
+
+        verify(cardDelegate).onCleared()
+        verify(genericActionDelegate).onCleared()
+    }
+
+    @Test
+    fun `when observe is called then observe in delegates is called`() {
+        val lifecycleOwner = mock<LifecycleOwner>()
```

Comment:
Just an FYI there is a [TestLifecycleOwner](https://developer.android.com/reference/kotlin/androidx/lifecycle/testing/TestLifecycleOwner) from a testing library. It's overkill for this test though, but just wanted to let you know

Reply:
👌